packages/flutter/test/widgets/sensitive_content_test.dart

```diff
@@ -0,0 +1,1414 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  // Default content sensitivity setting for testing.
+  final int defaultContentSensitivitySettingId = ContentSensitivity.autoSensitive.id;
+  final ContentSensitivity defaultContentSensitivitySetting =
+      ContentSensitivity.getContentSensitivityById(defaultContentSensitivitySettingId);
+
+  // The state of content sensitivity in the app.
+  final SensitiveContentSetting sensitiveContentSetting = SensitiveContentSetting.instance;
+
+  // The number of method channel calls to `SensitiveContent.setContentSensitivity`.
+  int setContentSensitivityCallCount = 0;
```

Comment:
Nit: Instead of a count, you could do a List<ContentSensitivity>, where the length of the list is the same as the count. Then you could also test what it gets called with. Not sure if this is actually valuable or not here though, up to you.

Reply:
Love this idea and I think it's useful! Helps us verify what we actually requested the native side set. Done.